dataflow/src/org/checkerframework/dataflow/util/PurityChecker.java

```diff
@@ -357,6 +388,10 @@ public PurityResult visitMethodInvocation(MethodInvocationTree node, PurityResul
                     p.addNotDetReason(node, reason);
                 } else if (!seFree) {
                     p.addNotSeFreeReason(node, reason);
+                } else if (!sdet) {
```

Comment:
I did try setting branches completely separate from the already implemented deterministic annotation, and I still had the same problem. I will update you once I completely remove the already existing annotation and replace all the current tests as well.